ospf6d/ospf6_nssa.c

```diff
@@ -177,16 +178,21 @@ void ospf6_abr_nssa_check_status(struct ospf6 *ospf6)
 				}
 			}
 		}
+
+		/* RFC3101, 3.1:
+		 * All NSSA border routers must set the E-bit in the Type-1
+		 * router-LSAs of their directly attached non-stub areas, even
+		 * when they are not translating.
+		 */
+		if (old_state != area->NSSATranslatorState) {
```

Comment:
If I have 3 areas and 2 of them the old state is not equal to the area->NSSATranslatorState we need to update the asbr status multiple times?  Why are we doing that?  Shouldn't we gather the data and do it once outside the loop ?

Reply:
It got merged already...
Anyway, the main fix here is to correctly calculate `ospf6->redist_count` which should be incremented/decremented only when an area actually changed its state.
I agree that we could move the call to an `ospf6_asbr_status_update` to the end of the function, but there's no harm in calling it every time because it returns right away if the status is already the same as desired.